### PR TITLE
ENT-4279: Switch away from rhsmlib.compat.subprocess_compat

### PR DESCRIPTION
**Pull request by m-horky**
* Card ID: ENT-4279

'subprocess_compat()' was required for Python < 2.7, as its version of
'subprocess' did not have the 'check_output()' function.

subscription-manager has been Python 3+ for some time now, this
compatibility wrapper can be safely removed.

### CODE REVIEW

**Comment by ptoscano** [comment]
Hmm IIRC that old wrapper returned `str` while `subprocess.check_output()` return `bytes`, so better test that the facts generation still works properly by running the "facts" subcommand of sub-man.

Also, please test the old code path in src/rhsmlib/facts/hwprobe.py for non-JSON lscpu versions by forcing `_check_lscpu_json()` to False; that old code path is the one affected by this change (the JSON path already uses `subprocess.check_output()`).

**Comment by m-horky** [comment]
1/ I can confirm that the facts subcommand returns the same values as before this change.
2/ Non-JSON version returns less facts that the JSON one (`lscpu.` are missing), but the rest stays consistent.

File: src/rhsmlib/facts/virt.py [hunk 1]
```diff
@@ -34,7 +32,7 @@ def get_virt_info(self):
         virt_dict = {}
 
         try:
-            host_type = compat_check_output('/usr/sbin/virt-what')
+            host_type = subprocess.check_output('/usr/sbin/virt-what')
```
Comment (ptoscano) [inline src/rhsmlib/facts/virt.py:35]: this returns `bytes` now, so the `splitlines()` below will not work

File: src/rhsmlib/facts/hwprobe.py [hunk 2]
```diff
@@ -619,9 +615,8 @@ def _parse_lscpu_human_readable_output(self, lscpu_env):
         lscpu_cmd_string = ' '.join(lscpu_cmd)
 
         try:
-            lscpu_out = compat_check_output(lscpu_cmd,
-                                            env=lscpu_env)
-        except CalledProcessError as e:
+            lscpu_out = subprocess.check_output(lscpu_cmd, env=lscpu_env)
```
Comment (ptoscano) [inline src/rhsmlib/facts/hwprobe.py:618]: this returns `bytes` now, so the `strip().split('\n')` below will not work

**Review by ptoscano — CHANGES_REQUESTED** [review]
> 1/ I can confirm that the facts subcommand returns the same values as before this change.

In non-virtualized guests yes, as `virt-what` returns nothing. However, due to `bytes` vs `str`, the "virt" facts are broken on virtual guests.

> 2/ Non-JSON version returns less facts that the JSON one (lscpu. are missing), but the rest stays consistent.

Less facts because of another `bytes` vs `str` issue :-(

File: src/rhsmlib/facts/virt.py (hunk 1 above)
Comment (m-horky) [reply src/rhsmlib/facts/virt.py:35]: I should have read properly and completely what the `compat_check_output` did. Thanks.

**Comment by m-horky** [comment]
Thanks for the review again, it should be fixed now.

File: src/rhsmlib/facts/hwprobe.py [hunk 3]
```diff
@@ -619,9 +615,9 @@ def _parse_lscpu_human_readable_output(self, lscpu_env):
         lscpu_cmd_string = ' '.join(lscpu_cmd)
 
         try:
-            lscpu_out = compat_check_output(lscpu_cmd,
-                                            env=lscpu_env)
-        except CalledProcessError as e:
+            lscpu_raw_out: bytes = subprocess.check_output(lscpu_cmd, env=lscpu_env)
+            lscpu_out: str = lscpu_raw_out.decode("utf-8").strip()
```
Comment (ptoscano) [inline src/rhsmlib/facts/hwprobe.py:619]: the `strip()` here is not needed, as it is done few lines below

File: src/rhsmlib/facts/virt.py [hunk 4]
```diff
@@ -34,7 +32,8 @@ def get_virt_info(self):
         virt_dict = {}
 
         try:
-            host_type = compat_check_output('/usr/sbin/virt-what')
+            host_type_raw: bytes = subprocess.check_output('/usr/sbin/virt-what')
+            host_type: str = host_type_raw.decode("utf-8").strip()
```
Comment (ptoscano) [inline src/rhsmlib/facts/virt.py:36]: i think there is no need for `strip()` here, as it was not done before

**Review by ptoscano — CHANGES_REQUESTED** [review]
> Thanks for the review again, it should be fixed now.

Yes indeed, all the facts work fine again, in virtual and non-virtual cases, and in texual lscpu and JSON lscpu. Thanks for the updates (even to the tests)!

I just left a couple of very minor nits to adjust; after this, it is ready to go in IMHO.

File: src/rhsmlib/facts/virt.py (hunk 4 above)
Comment (m-horky) [reply src/rhsmlib/facts/virt.py:36]: You are right; the `striplines()` below it can deal with trailing newline.

**Review by ptoscano — APPROVED** [review]
Thanks for the fixes! LGTM now

**Pull request merged**